src/update-function/swap-array-control.ts

```diff
@@ -0,0 +1,24 @@
+import { SwapArrayControlAction } from '../actions';
+import { formArrayReducer } from '../array/reducer';
+import { FormArrayState } from '../state';
+import { ensureState } from './util';
+
+/**
+ * This update function takes a source index, a destination index, and returns a projection function
```

Comment:
typo: "[...] a source index and a destination index, and returns [...]" (even though the comma before the second "and" is not 100% correct it helps with readability in my opinion)